packages/react-component-library/src/components/AutocompleteE/AutocompleteEOption.tsx

```diff
@@ -3,11 +3,28 @@ import React from 'react'
 import { SelectBaseOption, SelectBaseOptionProps } from '../SelectBase'
 
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface AutocompleteEOptionProps extends SelectBaseOptionProps {}
+export interface AutocompleteEOptionProps extends SelectBaseOptionProps {
+  inputValue?: string
+}
+
+function emboldenInputValue(children: string, inputValue: string) {
+  if (inputValue) {
+    const regex = new RegExp(`(${inputValue})`, 'gi')
+    return children.replace(regex, '<strong>$1</strong>')
+  }
+
+  return children
+}
 
 export const AutocompleteEOption = React.forwardRef<
   HTMLLIElement,
   AutocompleteEOptionProps
->((props, ref) => <SelectBaseOption {...props} ref={ref} />)
+>(({ children, inputValue, ...rest }, ref) => {
+  return (
+    <SelectBaseOption ref={ref} {...rest}>
+      {emboldenInputValue(children, inputValue)}
```

Comment:
Less important than the other comments, but if it's HTML I think it would be better to pass it as e.g. an `html` prop as you wouldn't normally expect `children` to contain raw HTML. (Alternatively could try to rejig things to use `dangerouslySetInnerHTML` here instead of in `SelectBaseOption`.)

Reply:
This is a good idea as I felt uncomfortable just applying it to `children`.